files/dns/nginx-vhost

```diff
@@ -0,0 +1,27 @@
+# This file is managed by Juju and follows the Template Language from Consul:
+# https://github.com/hashicorp/consul-template#templating-language
+
+{{range services}}
+upstream {{.Name}} {
+    least_conn;
+    {{range service .Name}} server {{.Address}}:{{.Port}} max_fails=3 fail_timeout=60 weight=1;
+    {{end}}
+}
+
+
+server {
+  listen 80 {{.Name}}.domain;
+
+  charset utf-8;
+
+  location / {
```

Comment:
This is the "server" config, which denotes which DNS entry to listen for, and which upstream block to consume.